lib/api/model/narrow.dart

```diff
@@ -1,5 +1,18 @@
 typedef ApiNarrow = List<ApiNarrowElement>;
 
+/// Resolve any [ApiNarrowDm] elements appropriately.
+///
+/// This encapsulates a server-feature check.
+ApiNarrow resolveDmElements(ApiNarrow narrow, int zulipFeatureLevel) {
+  final supportsOperatorDm = zulipFeatureLevel >= 177; // TODO(server-7)
+  return (narrow.any((element) => element is ApiNarrowDm))
+    ? narrow.map((element) => switch (element) {
+        ApiNarrowDm() => element.resolve(legacy: !supportsOperatorDm),
+        _             => element,
+      }).toList()
+    : narrow;
```

Comment:
This feels a bit tangly to read, as I think often happens with the `?:` operator when one or both branches are more than a line or two long. Let's leave it as full statement-level control flow, like it is in the existing code:
```suggestion
  if (!narrow.any((element) => element is ApiNarrowDm)) {
    return narrow;
  }
  final supportsOperatorDm = zulipFeatureLevel >= 177; // TODO(server-7)
  return narrow.map((element) => switch (element) {
    ApiNarrowDm() => element.resolve(legacy: !supportsOperatorDm),
    _             => element,
  }).toList();
```